### PR TITLE
feat: add an endpoint to edit posts

### PR DESCRIPTION
**Pull request by muzzacode**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/auth-impl/src/main/java/com/datahub/authentication/post/PostService.java [hunk 1]
```diff
@@ -87,4 +92,35 @@ public boolean deletePost(@Nonnull Urn postUrn, @Nonnull Authentication authenti
...
   }
+
+  public boolean updatePost(
+      @Nonnull Urn postUrn,
+      @Nonnull String postType,
+      @Nonnull PostContent updatedContent,
+      @Nonnull Authentication authentication)
+      throws RemoteInvocationException, URISyntaxException {
+
+    val response =
```
Comment (anshbansal) [inline metadata-service/auth-impl/src/main/java/com/datahub/authentication/post/PostService.java:103]: Please use `final EntityResponse entityResponse` instead. We use specific types in the java code base

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 2]
```diff
@@ -582,6 +582,15 @@ type Mutation {
     """
     createPost(input: CreatePostInput!): Boolean
 
+    """
+    Update a post
+    """
+    updatePost(
+        "The urn of the View to update"
+        urn: String!,
```
Comment (anshbansal) [inline datahub-graphql-core/src/main/resources/entity.graphql:590]: Please move `urn` inside `UpdatePostInput`  type and make changes on java side accordingly. This is for consistency with other graphql endpoint Inputs

**Review by anshbansal — CHANGES_REQUESTED** [review]

File: metadata-service/auth-impl/src/main/java/com/datahub/authentication/post/PostService.java (hunk 1 above)
Comment (muzzacode) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/post/PostService.java:103]: sure, it is just to make it less verbose. But I will add explicit type here.

File: datahub-graphql-core/src/main/resources/entity.graphql (hunk 2 above)
Comment (muzzacode) [reply datahub-graphql-core/src/main/resources/entity.graphql:590]: Sure, I will move it though I followed the existing _UpdateViewInput_ type.

**Review by anshbansal — APPROVED** [review]

**Pull request merged**